### ISSUE TITLE
Can not reproduce softmax+triplet loss result

### ISSUE THREAD

**Issue by cptay** [KaiyangZhou/deep-person-reid#330]
Hi,

Thanks for sharing your codes.

I tried to train on Market1501 using osnet_x1_0 with the following setting
- loss = 1.0 * cross entropy + 0.5 * triplet
- cosine annealing learning, max epoch = 250, lr = 0.015
- backbone weight freezed for first 10 epoch
- smooth labeling, cosine dist metric
- seed = 1
- batch 16x4, image size 256 x 128

The accuracy result was mAP 87.1%, CMC 95.1%. The CMC result was 0.4% lower than what was stated in the paper. Is this expected? How to improve it further?

Thanks

**Comment by KaiyangZhou** [comment]
it's normal

try tuning the training params which may give you better performance

you may want to look at the generalization performance (cross-dataset) as well for evaluation

**Comment by cptay** [comment]
Can you share with us the seed numbers you used in your tests?

**Comment by KaiyangZhou** [comment]
I normally use `seed=1,2,3,...` unless sometimes getting crazy and put random numbers like 9485, 2837, ... (via random typing)

NOTE that even if the initialization params of the network are kept the same due to the fixed seed (this is however not guaranteed if the software versions are different), the batch sampling can vary during the course of training which will lead to different local optima and hence different results (BUT definitely within a reasonable range)

btw, your `95.1%` should be considered to be a good reproduction of the reported `95.5%+-0.1` so the title "cannot reproduce" is a false claim :)

**Comment by cptay** [comment]
I am not trying to be nitpicking but simply wanted a training method to get to the 95%, or even 97% accuracy in CMC.

My interpretation of the stated 95.5%+-0.1 translates to 95.4% to 95.6%. To get from 95% to 95.5% is no trivial task. Regardless, I will train with two different seeds and tabulate the mean. Hopefully, I get better results.

BTW, I did try with seed 1,2,3,4 using osnet_x1_0, but without label smoothing and cosine dist metric.
The results were

86.4, 95.2
86.3  95.2
86.4  95.0
86.3  94.9

There is obvious improvement in mAP accuracy with label smoothing and cosine dist metric, but not so sure about Rank 1

**Comment by tomFoxxxx** [comment]
> Hi,
> 
> Thanks for sharing your codes.
> 
> I tried to train on Market1501 using osnet_x1_0 with the following setting
> 
> * loss = 1.0 * cross entropy + 0.5 * triplet
> * cosine annealing learning, max epoch = 250, lr = 0.015
> * backbone weight freezed for first 10 epoch
> * smooth labeling, cosine dist metric
> * seed = 1
> * batch 16x4, image size 256 x 128
> 
> The accuracy result was mAP 87.1%, CMC 95.1%. The CMC result was 0.4% lower than what was stated in the paper. Is this expected? How to improve it further?
> 
> Thanks

pretrained?
